Remotion/Validation/UnitTests/Rules/ObjectValidationRuleTest.cs

```diff
@@ -1,21 +1,176 @@
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Collections.Generic;
+using Moq;
+using NUnit.Framework;
+using Remotion.Validation.Results;
+using Remotion.Validation.Rules;
+using Remotion.Validation.Validators;
 
 namespace Remotion.Validation.UnitTests.Rules
 {
   [TestFixture]
   public class ObjectValidationRuleTest
   {
     [Test]
-    [Ignore("TODO RM-5906")]
-    public void Validate ()
+    public void Validate_ReturnsValidationFailures ()
     {
+      var context = new ValidationContext(new object());
+
+      var failure1 = new ObjectValidationFailure(
+          15,
+          "15 is not a number",
+          "Localized validation message: 15 is not a number.");
+
+      var failure2 = new ObjectValidationFailure(
+          42,
+          "42 is not a number",
+          "Localized validation message: 42 is not a number.");
+
+      var validator1Stub = new Mock<IObjectValidator>();
+      validator1Stub
+          .Setup(_ => _.Validate(It.IsAny<ObjectValidatorContext>()))
+          .Returns(new[] { failure1 });
+
+      var validator2Stub = new Mock<IObjectValidator>();
+      validator2Stub
+          .Setup(_ => _.Validate(It.IsAny<ObjectValidatorContext>()))
+          .Returns(new[] { failure2 });
+
+      var rule = new ObjectValidationRule<object>(null, new[] { validator1Stub.Object, validator2Stub.Object });
+
+      var result = rule.Validate(context);
+      Assert.That(result, Is.EquivalentTo(new[] { failure1, failure2 }));
+    }
+
+    [Test]
+    public void Validate_UsesCorrectChildContextToBuildParentContext ()
+    {
+      var instance = new object();
+      var validationContext = new ValidationContext(instance);
+
+      var failure1 = new ObjectValidationFailure(
+          15,
+          "15 is not a number",
+          "Localized validation message: 15 is not a number.");
+
+      var validator1Stub = new Mock<IObjectValidator>();
+      validator1Stub
+          .Setup(_ => _.Validate(It.Is<ObjectValidatorContext>(ctx => ctx.ParentContext == validationContext && ctx.Instance == instance)))
+          .Returns(new[] { failure1 });
+
+      var rule = new ObjectValidationRule<object>(null, new[] { validator1Stub.Object });
+
+      var result = rule.Validate(validationContext);
+      Assert.That(result, Is.EquivalentTo(new[] { failure1 }));
+    }
+
+    [Test]
+    public void Validate_WithNoValidators_ReturnsEmptyResult ()
+    {
+      var context = new ValidationContext(new object());
+      var rule = new ObjectValidationRule<object>(null, new List<IObjectValidator>());
+
+      var result = rule.Validate(context);
+
+      Assert.That(result, Is.Empty);
+    }
+
+    [Test]
+    public void Validate_WithNullAsInstanceToValidate_ReturnsEmptyResultAsValidatorsAreNotCalled ()
```

Comment:
Tip: "AsValidatorsAreNotCalled` is something we usually put to the left or leave out. it's possible I wrote this myself during the review so...yeah. Nothing to fuss oer, though.